### PR TITLE
Specify behavior of int->float and f64->f32 casts

### LINKED ISSUE #570: Update undefined behavior note about float-to-float cast.
operator-expr.md mentions issue https://github.com/rust-lang/rust/issues/15536 as undefined behavior. I believe this is now fixed, and no longer undefined? Or something, someone should investigate and update the note.

### PR DESCRIPTION
**Pull request by rkruppe**
This documents the semantics proposed in https://github.com/rust-lang/rust/issues/62231

Closes #570

### CODE REVIEW

File: src/expressions/operator-expr.md [hunk 1]
```diff
@@ -368,15 +368,11 @@ same trait object.
...
+      representation of the integer, rounded if necessary (rounding to nearest,
+      ties to even)
     * Casting from an f32 to an f64 is perfect and lossless
     * Casting from an f64 to an f32 will produce the closest possible value
-      (rounding strategy unspecified)
-        * **[NOTE: currently this will cause Undefined Behavior if the value
-          is finite but larger or smaller than the largest or smallest finite
-          value representable by f32][float-float]**. This is a bug and will
-          be fixed.
+      (rounding to nearest, ties to even)
```
Comment (dhardy) [inline src/expressions/operator-expr.md]: Might be useful to mention saturation to ±inf

**Comment by Havvy** [comment]
If we actually guarantee it, the guarantee should be approved by the lang team first.

**Comment by gnzlbg** [comment]
We can't ping the lang team here because this repo is part of the nursery and not of the rust-lang org. cc @Centril

**Comment by Centril** [comment]
An issue should be made on rust-lang/rust, tagged with t-lang + description of what guarantee is sought, and then we can consider that.

**Comment by rkruppe** [comment]
Finally got around to doing that: https://github.com/rust-lang/rust/issues/62231

**Comment by rkruppe** [comment]
Updated with explicit mention of overflow behavior, better formatting, and the note about performance requested by @joshtriplett.

File: src/expressions/operator-expr.md [hunk 2]
```diff
@@ -385,8 +384,19 @@ same trait object.
...
 
+\* if integer-to-float casts with this rounding ode and overflow behavior are
+not supported natively by the hardware, these casts will likely be slower than
+expected.
+
+\*\* if f64-to-f32 casts with this rounding mode and overflow behavior are not
+supported natively by the hardware, these casts will likely be slower than
+expected.
+
+\*\*\* as defined in IEEE 754-2008 &sect;4.3.1: pick the nearest floating point
```
Comment (Centril) [inline src/expressions/operator-expr.md]: Nit: might be a bit more readable with numbered footnotes?

Comment (rkruppe) [reply src/expressions/operator-expr.md]: I agree, but this style of footnotes is used elsewhere in this chapter and in some other chapters too, and both deviating locally and updating them all seems ungreat.

**Comment by rkruppe** [comment]
(Force-pushed just to edit the commit message: accidentally self-referenced this PR instead of pointing at https://github.com/rust-lang/rust/issues/62231)

File: src/expressions/operator-expr.md [hunk 3]
```diff
@@ -385,8 +384,19 @@ same trait object.
 * `u8` to `char` cast
     * Casts to the `char` with the corresponding code point.
 
+\* if integer-to-float casts with this rounding ode and overflow behavior are
```
Comment (joshtriplett) [inline src/expressions/operator-expr.md]: s/ode/mode/

**Comment by joshtriplett** [comment]
One minor typo; other than that, this looks ready to go.

File: src/expressions/operator-expr.md (hunk 2 above)
Comment (Centril) [reply src/expressions/operator-expr.md]: Oh I see; guess I didn't search thoroughly enough to tell what the overall style is. :)

**Comment by Centril** [comment]
FCP completed in https://github.com/rust-lang/rust/issues/62231#issuecomment-529232881.

**Pull request merged**